src/NServiceBus.Core/CircuitBreakers/ConsecutiveFailuresCircuitBreaker.cs

```diff
@@ -0,0 +1,66 @@
+namespace NServiceBus
+{
+    using System;
+    using System.Threading;
+    using System.Threading.Tasks;
+    using Logging;
+
+    class ConsecutiveFailuresCircuitBreaker : IDisposable, ICircuitBreaker
+    {
+        public ConsecutiveFailuresCircuitBreaker(string name, int consecutiveFailuresBeforeTriggering, Action<Exception, long> triggerAction, Action<long> disarmAction, TimeSpan armedFailureDelayDuration)
+        {
+            this.name = name;
+            this.triggerAction = triggerAction;
+            this.consecutiveFailuresBeforeTriggering = consecutiveFailuresBeforeTriggering;
+            this.disarmAction = disarmAction;
+            this.armedFailureDelayDuration = armedFailureDelayDuration;
+        }
+
+        public void Success()
+        {
+            var oldValue = Interlocked.Exchange(ref failureCount, 0);
+
+            if (oldValue >= consecutiveFailuresBeforeTriggering)
+            {
+                Logger.InfoFormat("The circuit breaker for {0} is now disarmed", name);
+                disarmAction(DateTime.Now.Ticks);
```

Comment:
I guess for this purpose it doesn't really matter whether you use DateTime.Now or UtcNow yet I think it would still be better to use the best practice way. 

In theory, you wouldn't even have to use the tick as long as you call disarm and trigger action. It looks almost like some logic that should be in the receive component has leaked into this class.